backend/app/models.py

```diff
@@ -44,5 +58,285 @@ class Document(models.Model):
     title = models.CharField(max_length=255, blank=True)
     word_count = models.PositiveIntegerField(blank=True, null=True, default=None)
     tokenized_text = models.JSONField(null=True, blank=True, default=None)
-    word_counts_counter = models.JSONField(null=True, blank=True, default=dict)
+    word_count_counter = models.JSONField(null=True, blank=True, default=dict)
     part_of_speech_tags = models.JSONField(null=True, blank=True, default=list)
+
+    objects = DocumentManager()
+
+    def _clean_quotes(self):
+        """
+        Scans through the text and replaces all of the smart quotes and apostrophes with their
+        "normal" ASCII variants
+
+        :param self: The Document to reformat
+        :return: A string that is identical to `text`, except with its smart quotes exchanged
+        """
+
+        smart_quotes = {
+            '“': '"',
+            '”': '"',
+            "‘": "'",
+            "’": "'",
+        }
+
+        output_text = self.text
+        for quote in smart_quotes:
+            output_text = output_text.replace(quote, smart_quotes[quote])
+        self.text = output_text
+        self.save()
+        return self.text
+
+    def get_tokenized_text(self):
+        """
+        Tokenizes the text of a Document and returns it as a list of tokens, while removing all punctuation
+        and converting everything to lowercase.
+
+        :param self: The Document to tokenize
+        :return: List of each word in the Document
+        """
+
+        if self.tokenized_text is None:
+            self._clean_quotes()
+            tokens = nltk.word_tokenize(self.text)
+            excluded_characters = set(string.punctuation)
+            tokenized_text = [word.lower() for word in tokens if word not in excluded_characters]
+            self.tokenized_text = tokenized_text
+            self.save()
+        return self.tokenized_text
+
+    def get_count_of_word(self, word):
+        """
+        Returns the number of instances of a word in the text.
+
+        Note: This method is not case sensitive
+
+        :param word: word to be counted in text
+        :return: Number of occurrences of the word, as an int
+        """
+        self.get_word_count_counter()
+        return self.word_count_counter[word.lower()]
```

Comment:
Per my comment above about streamlining these methods, if we followed that plan, this method becomes:
```
return JSON.loads(self.word_count_counter)[word.lower()]
```
Or something like that.

Reply:
fixed!